### DISCUSSION TITLE
I think there's a problem with brotli

### DISCUSSION THREAD

**Discussion by garfieldairlines** [ZoeyVid/NPMplus#722] [Q&A] [answered]
Hi,
I had a lot of issues with my internal webpages, the css and js wasn't loading… I thought it was many things (http3 ? certificate ?) but after a closer look it's according to me, probably brotli.

The following configuration in nginx/conf/conf.d/include/brotli.conf doesn't bring a lot : 
```
brotli on;
brotli_types *;
brotli_comp_level 0;
brotli_static on;
```

First, it encompasses all types and compresses everything (including formats like eventually music or webp which are already compressed), then comp_level 0 brings close to nothing, and finally brotli_static on is only a advantage if your server already has files in .br already stored in the same folder. (Understand : if you have a index.html.br next to index.html, nginx will directly deliver that file and won't compress on the fly).

If I may, here's my suggestion for a better config snippet : 

```
brotli						on;
brotli_comp_level			8;
brotli_types				application/atom+xml application/javascript application/json application/vnd.api+json application/rss+xml application/vnd.ms-fontobject application/x-font-opentype application/x-font-truetype application/x-font-ttf application/x-javascript application/xhtml+xml application/xml application/octet-stream application/x-web-app-manifest+json application/xml+rss font/eot font/opentype font/otf font/truetype font/woff font/woff2 image/svg+xml image/vnd.microsoft.icon image/bmp image/x-icon image/x-win-bitmap text/css text/javascript text/plain text/xml text/x-component;
```
If you want, a ratio of 6 (which is recommended) is a better trade off between compression and CPU load.

I am currently trying it via the file custom/server_proxy.conf and so far I don't see any problems.

**Comment by garfieldairlines** [comment]
I'd also add an edit for gzip from : 
```
    gzip on;
    gzip_vary on;
    gzip_types *;
    gzip_proxied any;
    gzip_comp_level 1;
    gunzip on;
    gzip_static on;
```
To
```
gzip on;
gzip_vary on;
gzip_proxied expired no-cache no-store private auth;
gzip_comp_level 8;
gzip_buffers 16 8k;
gzip_http_version 1.1;
gzip_types application/atom+xml application/javascript application/json application/vnd.api+json application/rss+xml application/vnd.ms-fontobject application/x-font-opentype application/x-font-truetype application/x-font-ttf application/x-javascript application/xhtml+xml application/xml application/octet-stream application/x-web-app-manifest+json application/xml+rss font/eot font/opentype font/otf font/truetype font/woff font/woff2 image/svg+xml image/vnd.microsoft.icon image/bmp image/x-icon image/x-win-bitmap text/css text/javascript text/plain text/xml text/x-component;
```

**Comment by Zoey2936** [comment, marked as answer]
Hello, maybe I explain why I use this options:
brotli_static on; : doesn't hurt, if someone provides this files it will use them, if not it won't use them
brotli on; : self explaining
brotli_comp_level 0; : maybe read this PR comments: https://github.com/nextcloud/all-in-one/pull/2576 and this: https://blog.cloudflare.com/results-experimenting-brotli
brotli_types *; level 0 is very fast compression and compression is only done if a client requests it, so compression can be applied to anything
